### PR TITLE
Issue create success check middleware

### LINKED ISSUE #635: Create a success check middleware
## Is your feature request related to a problem? Please describe

A lot of apis provide a status field, we should provide a standard implementation to handle the most common patterns.

## Describe the solution you'd like

- success middleware(or):
  - status field is success
  - success field is true
- failure middleware(or):
  - status field is error
  - success field is false

### PR DESCRIPTION
**Pull request by owais34**
# The Pull Request is ready

- [x] fixes #635 
- [x] all actions are passing
- [x] only fixes a single issue

## Overview

The following two standards for api response have been taken as reference for this new feature

1) JSEND - https://github.com/omniti-labs/jsend

  {
    "data":{}
    "status": "fail/error",
    "message":"Present only when status is error"
  }

2) Alternative Standard Format 
{
  "data":{}
  "success": false,
  "message": "Used mostly in case of error message",
  "error_code": 1308,
}

Hence two middlewares were created : 

1. Success Check : Checks whether response has "status"/"success" field , with values describing a successful response , throws an error otherwise.
2. Failure Check : Checks whether response has "status"/"success" field , with values describing a failure response , throws an error otherwise.

## Review points

Need review whether the changes made , were in accordance with the problem statement.

## Framework

- [ ] the change breaks no interface
- [ ] default behaviour did not change
- [ ] any new text output is added to the translation files (at least the english one)
- [ ] tests have been added (if required)
- [ ] documentation has been adjusted (if required)
- [ ] shared code has been extracted in a different file

## Summary by CodeRabbit

- **New Features**
	- Introduced middleware classes `FailureCheck` and `SuccessCheck` for enhanced API response handling, ensuring error and success scenarios are properly validated.
	- Added `StandardResponse` interface to maintain a consistent structure for response objects.
- **Documentation**
	- Updated error messages to align with new response validation capabilities.

### CODE REVIEW

**Review by Idrinth — CHANGES_REQUESTED** [review]
please add the middlewares to the middleware list on the documentation page

**Comment by owais34** [comment]
@Idrinth  , I'll have resolve the suggestions provided by [coderabbitai](https://github.com/apps/coderabbitai) ?
Thanks for the appreciation !!

**Comment by Idrinth** [comment]
> @Idrinth , I'll have resolve the suggestions provided by [coderabbitai](https://github.com/apps/coderabbitai) ? Thanks for the appreciation !!

Only if they make sense. The main issue is, that tests are failing. I believe en.yml is missing a few new translations ;)

File: framework/src/middlewares/success-check.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      throw new Error(
+        language('response_not_success', 'status', `${ response.status }`,),
+      );
+    }
+
+    if (response.success === false) {
+      throw new Error(
+        language('response_not_success', 'success', `${ response.success }`,),
+      );
+    }
```
[the remark this answers is not in the document]
Comment (owais34) [reply framework/src/middlewares/success-check.ts:51]: NA

File: framework/language/en.yml [hunk 2]
```diff
@@ -39,3 +39,5 @@ invalid_post_definition: The definition of post on route %1% is not valid.
 unknown_route_property: The property %1% on route %1% is unknown.
 invalid_request_property: The request property %2% of task %1% is invalid.
 invalid_request: The request task %1% has unknown properties.
+response_not_success: 'The response was not success, %1% field was %2%'
+response_not_failure: 'The response was not failure, %1% field was %2%'
```
[the remark this answers is not in the document]
Comment (Idrinth) [reply framework/language/en.yml:43]: please address this

File: framework/src/middlewares/failure-check.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+2) Another Common Format
+{
+  "success": false,
+  "message": "Used mostly in case of error message,
+  "error_code": 1308,
+}
+*/
+const SUCCESS = 'success';
```
Comment (Idrinth) [inline framework/src/middlewares/failure-check.ts:24]: please just integrate the string, single use constants for strings have little benefit.

File: framework/src/middlewares/success-check.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+2) Another Common Format
+{
+  "success": true,
+  "data": {}
+}
+*/
+
+const FAIL = 'fail';
+const ERROR = 'error';
```
Comment (Idrinth) [inline framework/src/middlewares/success-check.ts:25]: please just integrate the string, single use constants for strings have little benefit.

File: framework/test/middlewares/failure-check.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/* eslint-disable max-len */
```
Comment (Idrinth) [inline framework/test/middlewares/failure-check.ts:1]: please make thhis line specific

**Review by Idrinth — CHANGES_REQUESTED** [review]

**Review by Idrinth — APPROVED** [review]

**Pull request merged**